### ISSUE TITLE
PING Privacy Review: Other browser implementations

### ISSUE THREAD

**Issue by kdeqc** [WICG/floc#70]
For other browsers to implement FLoCs, they'd need to also store data that they don't necessarily do now.  Google is initially looking at domains only here - which I think is data that's available in all browsers, but I can't say that for sure.  Has this issue been considered?

Also, browsers having vastly different algorithms for something like FLoC decreases standardization and will likely make developers be more browser specific in how they handle things.  That might be unavoidable, but it would be good if there could be some level of cross-browser agreement about the data/algorithms involved.

**Comment by jkarlin** [comment]
Thanks for taking a look and opening the issue!

> For other browsers to implement FLoCs, they'd need to also store data that they don't necessarily do now. Google is initially looking at domains only here - which I think is data that's available in all browsers, but I can't say that for sure. Has this issue been considered?

The only other data that Chrome has considered thusfar besides domains is a categorization of the site being visited. Which I don't believe any browser has the capability of at the moment. So you make a good point. Our intent is to make any models that we produce available for other browsers to use as well.

> Also, browsers having vastly different algorithms for something like FLoC decreases standardization and will likely make developers be more browser specific in how they handle things. That might be unavoidable, but it would be good if there could be some level of cross-browser agreement about the data/algorithms involved.

We're still at the point of determining if this is even feasible (hence the Origin Trial) and we'll likely need to explore a number of algorithms. We'd love to end up in a state where we have cross-browser agreement on the data and algorithms. We're open for discussion interested browsers. And we'll certainly publish all of our methods so that other browsers can determine if they're a good fit and use them if they like them.

**Comment by PanMat** [comment]
Unless I am missing something here this seems flawed and very unfair to
smaller publishers. There is little to no value to derive from knowing that
a user visited YouTube or Twitch compared to knowing that a user visited a
niche site like ketodiets.com or AspenRealEstate.com or NapaWines.com.

**Comment by michaelkleber** [comment]
@PanMat Yup, there has been a bunch of discussion of this in #45.  We would definitely like to use a clustering technique that got more information from large heterogeneous sites, it's just ongoing research, as it's harder than the domain-only approach that we were able to implement as a first pass.
